src/package.rs

```diff
@@ -385,16 +363,44 @@ impl TryFrom<Bytes> for Package {
 
         Ok(Self { manifest, tgz })
     }
+
+    /// The name of this package
+    #[inline]
+    pub fn name(&self) -> &PackageName {
+        &self.manifest.package.name
+    }
+
+    /// The version of this package
+    #[inline]
+    pub fn version(&self) -> &Version {
+        &self.manifest.package.version
+    }
+
+    /// Lock this package
+    ///
+    /// Note that despite returning a Result this function never fails
+    pub fn lock(
+        &self,
+        registry: RegistryUri,
+        repository: String,
+        dependants: usize,
+    ) -> miette::Result<LockedPackage> {
+        LockedPackage::lock(self, registry, repository, dependants)
+    }
 }
 
-impl From<&Package> for Bytes {
```

Comment:
Why has this been removed?

Reply:
It is a red flag to have a `From` for a field unless the field semantically represents that type. A package is a package, not something that is equal to some bytes. Also, most importantly, it was not used. Better to use `package.tgz` explicitly.

Reply:
A package is strictly equal to bytes

Reply:
So, you can do `Bytes -> Package -> Bytes -> Package -> Bytes` all day

The manifest is only parsed out in the `TryFrom` to prevent the need of ungzipping on every manifest access